atlp-rwanda/winners-c8-bn-be
# #182209984 Set up an empty Express Boilerplate with dotenv

#### What does this PR do?
It sets up a hierarchy of directories that developers will need to work on and basic files(like `package.json`). It includes folders like database, controllers, validations and routes.
#### Description of Task to be completed?
Have the following aspects present:
 1. Set up the basic directories' hierarchy
 2. Set up basic npm scripts working (build, start, ...)
 3. Set up basic npm modules (babel, dotenv, ...) & ensure that they're working.
#### How should this be manually tested?
 1. Clone the repository
 2. Run `npm install`
 3. Edit the `.env` file to contain the needed variables, and edit any JS file under `src` folder to use the variables using `dotenv`
 4. Run `npm run build`
 5. Run `npm start` 
 6. The following endpoints should work:
     `GET http://localhost:5000/` shall return `{"message":"Hello World!"}`
     `GET http://localhost:5000/api/users` shall return `{"message":"Nothing is set yet!"}`
#### Any background context you want to provide?
The initial file in `/src/database/models/index.js` does not use ES6 because it was auto generated through `npx sequelize-cli init` command. This should be dealt by the developer. (UPDATE: I have commented its javascript codes so that the HoundCI will not keep indicating errors from that file due to ES6 syntax related problems)
#### What are the relevant pivotal tracker stories?
#182209984
